lib/backup/azremote/azblob.go

```diff
@@ -178,12 +178,35 @@ func (fs *FS) CopyPart(srcFS common.OriginFS, p common.Part) error {
 		return fmt.Errorf("failed to generate SAS token of src %q: %w", p.Path, err)
 	}
 
-	// Hotfix for SDK issue: https://github.com/Azure/azure-sdk-for-go/issues/19245
-	t = strings.Replace(t, "/?", "?", -1)
 	ctx := context.Background()
-	_, err = dbc.CopyFromURL(ctx, t, &blob.CopyFromURLOptions{})
+
+	// In order to support copy of files larger than 256MB, we need to use the async copy
+	// Ref: https://learn.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url
+	_, err = dbc.StartCopyFromURL(ctx, t, &blob.StartCopyFromURLOptions{})
 	if err != nil {
-		return fmt.Errorf("cannot copy %q from %s to %s: %w", p.Path, src, fs, err)
+		return fmt.Errorf("cannot start async copy %q from %s to %s: %w", p.Path, src, fs, err)
+	}
+
+	var copyStatus *blob.CopyStatusType
+	var copyStatusDescription *string
+	for {
+		r, err := dbc.GetProperties(ctx, nil)
+		if err != nil {
+			return fmt.Errorf("failed to check copy status, cannot get properties of %q at %s: %w", p.Path, fs, err)
+		}
+
+		// After the copy will be finished status will be changed to success/failed/aborted
+		// Ref: https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-properties#response-headers - x-ms-copy-status
+		if *r.CopyStatus != blob.CopyStatusTypePending {
+			copyStatus = r.CopyStatus
+			copyStatusDescription = r.CopyStatusDescription
+			break
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
Probably, it's possible to remove sleep and use some polling/watching for changes?

Reply:
Unfortunately API has no method which implements that.